torch/csrc/jit/tensorexpr/bounds_overlap.cpp

```diff
@@ -71,6 +71,83 @@ OverlapKind boundOverlap(Bound a, Bound b) {
   return PartialOverlap;
 }
 
+bool compareBound(
+    const Bound& a,
+    const Bound& b,
+    BoundCompareResult* cmp_result_ptr) {
+  TORCH_INTERNAL_ASSERT(cmp_result_ptr != nullptr);
+
+  if (a == b) {
+    *cmp_result_ptr = BoundCompareResult::kEQ;
+    return true;
+  }
+
+  auto overlap_kind = boundOverlap(a, b);
+  if (overlap_kind == OverlapKind::PartialOverlap) {
+    // The element of a and b could not always satisfy the comparison
+    //    a = [2, 10], b = [5, 15]
+    return false;
```

Comment:
For partial overlaps such as a = [2, 5], b = [5, 10], we still can compare them right? Such cases correspond to kLE or kGE.

Reply:
> For partial overlaps such as a = [2, 5], b = [5, 10], we still can compare them right? Such cases correspond to kLE or kGE.

Oh, yes. Updated the code.